File: 150_Evaluate_Reverse_Polish_Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stackOfNumbers = [] 
        for token in tokens: 
            if token not in ['+','*','/','-']:
                stackOfNumbers.append(int(token))
            else: 
                r = stackOfNumbers.pop()
                l = stackOfNumbers.pop() 
                
                if token == "+":
                    stackOfNumbers.append(r + l)
                elif token == "-":
                    stackOfNumbers.append(l - r) 
                elif token == "*":
                    stackOfNumbers.append(l * r) 
                else:
                    if l * r < 0 and l % r != 0: 
                        stackOfNumbers.append(l // r + 1) 
                    else:
                        stackOfNumbers.append( l // r) 
        return stackOfNumbers[-1]
                
                

File: 150_Evaluate_Reverse_Polish_Notation.java
import java.util.Stack; 
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stackOfNumbers = new Stack<Integer>(); 
        for(String s: tokens)
        {
            if(s.equals("+"))
            {
                stackOfNumbers.add( stackOfNumbers.pop() + stackOfNumbers.pop()); 
            }
            else if(s.equals("-"))
            {
                stackOfNumbers.add( -stackOfNumbers.pop() + stackOfNumbers.pop()); 
            }
            else if(s.equals("*"))
            {
                stackOfNumbers.add( stackOfNumbers.pop() * stackOfNumbers.pop()); 
            }
            else if (s.equals("/"))
            {
                int secondDigit = stackOfNumbers.pop();
                int firstDigit = stackOfNumbers.pop(); 
                stackOfNumbers.add(firstDigit / secondDigit); 
            }
            else
            {
                stackOfNumbers.add(Integer.parseInt(s)); 
            }
        }
        return stackOfNumbers.pop(); 
    }
}